pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoaderRegistry.java

```diff
@@ -37,14 +41,28 @@
  */
 public class SegmentDirectoryLoaderRegistry {
   private static final Logger LOGGER = LoggerFactory.getLogger(SegmentDirectoryLoaderRegistry.class);
-
   public static final String DEFAULT_SEGMENT_DIRECTORY_LOADER_NAME = "default";
-  private static final Map<String, SegmentDirectoryLoader> SEGMENT_DIRECTORY_LOADER_MAP = new HashMap<>();
+  private static final Map<String, SegmentDirectoryLoader> SEGMENT_DIRECTORY_LOADER_MAP = new ConcurrentHashMap<>();
```

Comment:
this change of using concurrent hash map may be kept

Reply:
done